lib/Vector2D.js

```diff
@@ -151,6 +151,8 @@ Vector2D.prototype.divide = function(scalar) {
  */
 Vector2D.prototype.angleBetween = function(that) {
     var cos = this.dot(that) / (this.length() * that.length());
+    if (cos < -1) cos = -1;
+    else if (cos > 1) cos = 1;
```

Comment:
alternately, I often use `cos = Math.max(-1, Math.min(1, cos));` but that does add some function call overhead